### PR TITLE
Question dashboard implementation

### PR DESCRIPTION
**Pull request by Sandipan-Ghosh**
**Fixed Issue**

#20900

**Files Changed**

CategoryController.cs -  Edited comments and implemented async-await.

QuestionController.cs - Edited comments and implemented async-await.

SingleMultipleAnswerQuestionAC.cs - Renamed the file from SingleMultipleAnswerQuestionApplicationClass to SingleMultipleAnswerQuestionAC

CategoryRepository.cs - Async-await is implemeted in GetAllCategories method and comments are edited

ICategoryrepository.cs - Edited comments and meaningful names are given

IQuestionRepository.cs - Edited comments

QuestionRepository.cs - Recreated the GetAllQuestions API

questions-dashboard.component.cs - Variable name changed

questions-dashboard.html.cs - Variable name changed

startup.cs - Removed extra spaces.

SingleMultipleAnswerQuestionAc- removed the file.

startup.cs - edited services.AddMvc , services.AddMvc( ).AddJsonOptions(options => options.SerializerSettings.ReferenceLoopHandling = Newtonsoft.Json.ReferenceLoopHandling.Ignore); .

SingleMultipleAnswerQuestion.cs - Rebuilt the relationship schema with Question.cs model

CodeSnippetQuestionModel.cs - Rebuilt the relationship schema with Question.cs model

**Checks**

- [x] Naming Conventions
- [x] Server side code comments (proper English, grammar and no spelling mistakes)
- [x] Client side code comments (proper English, grammar and no spelling mistakes)
- [x] Unused variables, methods, blank spaces and name spaces. 
- [x] Optimal Code and code formatting
- [x] Commit History is proper, no merge is done. 
- [x] Commit/pull request messages should be proper to justify your feature
- [ ] All test cases are executed provided by tester.

### CODE REVIEW

**Comment by Krunal-Promact** [comment]
Pull Request is not proper.

File: Trappist/src/Promact.Trappist.Core/Controllers/CategoryController.cs [hunk 1]
```diff
@@ -12,17 +12,17 @@ public CategoryController(ICategoryRepository categoryRepository)
         {
             _categoryRepository = categoryRepository;
         }
+        #region GetAllCategories
         /// <summary>
-        /// Get All Categories
+        /// The undermentioned controller calls the GetAllCategories method which is implemented in the CategoryRepository
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Core/Controllers/CategoryController.cs]: comment is not proper.

File: Trappist/src/Promact.Trappist.Core/Controllers/CategoryController.cs [hunk 2]
```diff
@@ -40,9 +40,7 @@ public IActionResult CatagoryAdd([FromBody] Category category)
             _categoryRepository.AddCategory(category);
             return Ok(category);
         }
-        #endregion
 
-        #region PutMethod
         /// <summary>
         /// Put Method
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Core/Controllers/CategoryController.cs]: Comment should be in one line.

File: Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs [hunk 3]
```diff
@@ -16,17 +17,25 @@ public QuestionRepository(TrappistDbContext dbContext)
...
-        public ICollection<SingleMultipleAnswerQuestionApplicationClass> GetAllQuestions()
+        /// The function name ends with Async
+        public async Task<ICollection<SingleMultipleAnswerQuestionAC>> GetAllQuestionsAsync()
         {
-            var questions = _dbContext.SingleMultipleAnswerQuestion.ProjectTo<SingleMultipleAnswerQuestionApplicationClass>().ToList();
-            questions.AddRange(_dbContext.CodeSnippetQuestion.ProjectTo<SingleMultipleAnswerQuestionApplicationClass>().ToList());
+            var questions = await _dbContext.Question.ProjectTo<SingleMultipleAnswerQuestionAC>().ToListAsync();
+
+            foreach (var question in questions)
+            {
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs]: please include method to get options value.

Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs]: please include a method to get options value.

File: Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/question.model.ts [hunk 4]
```diff
@@ -1,12 +1,25 @@
...
+//import { SingleMultipleAnswerQuestion } from '../questions/single-multiple-answer-question.model';
+class SingleMultipleAnswerQuestionOption {
+    option: string;
+    isAnswer: boolean;
+}
+//import { Question } from '../questions/question.model';
+//import { SingleMultipleAnswerQuestionOption } from '../questions/single-multiple-answer-question-option.model';
+class SingleMultipleAnswerQuestion {
+    id: number;
+    singleMultipleAnswerQuestionOption: SingleMultipleAnswerQuestionOption[] = new Array<SingleMultipleAnswerQuestionOption>();
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/question.model.ts]: initialization should go in constructor.

File: Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/question.model.ts [hunk 5]
```diff
@@ -1,12 +1,28 @@
...
+    option: string;
+    isAnswer: boolean;
+}
+
+class SingleMultipleAnswerQuestion {
+    singleMultipleAnswerQuestionOption: SingleMultipleAnswerQuestionOption[];
+    constructor() {
+        this.singleMultipleAnswerQuestionOption = new Array<SingleMultipleAnswerQuestionOption>();
+    }
+    id: number;
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/question.model.ts]: all properties should be declare first then constructor will be written.

File: Trappist/src/Promact.Trappist.DomainModel/Models/Question/CodeSnippetQuestion.cs [hunk 6]
```diff
@@ -1,11 +1,18 @@
...
-    public class CodeSnippetQuestion: BaseModel
+    public class CodeSnippetQuestion
     {
-        public int QuestionId { get; set; }
+        [ForeignKey("Question")]
+        public int Id { get; set; }
+
+        [Required]
+        public DateTime CreatedDateTime { get; set; }
+        public DateTime? UpdateDateTime { get; set; }
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/CodeSnippetQuestion.cs]: formatting is not proper.

File: Trappist/src/Promact.Trappist.DomainModel/Models/Question/SingleMultipleAnswerQuestion.cs [hunk 7]
```diff
@@ -1,15 +1,20 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
 using System.ComponentModel.DataAnnotations.Schema;
 
 namespace Promact.Trappist.DomainModel.Models.Question
 {
-    public class SingleMultipleAnswerQuestion: BaseModel
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/SingleMultipleAnswerQuestion.cs]: formatting is not proper.

File: Trappist/src/Promact.Trappist.Web/Startup.cs [hunk 8]
```diff
@@ -123,14 +107,6 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
             }
             context.Database.Migrate();
             context.Seed();
-            #region Auto Mapper Configuration
-            Mapper.Initialize(cfg =>
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/Startup.cs]: why auto mapper configurations removed?

Comment (Sandipan-Ghosh) [inline Trappist/src/Promact.Trappist.Web/Startup.cs]: As per the current model schema I don't need automapper,as I am not mapping any property of any model class to any application class in the server side, therefore I removed automapper configurations.

File: Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/question.model.ts [hunk 9]
```diff
@@ -1,12 +1,26 @@
...
+    singleMultipleAnswerQuestion: SingleMultipleAnswerQuestion;
+    codeSnippetQuestion: CodeSnippetQuestion;
+}
+
+class SingleMultipleAnswerQuestionOption {
+    option: string;
+    isAnswer: boolean;
+}
+
+class SingleMultipleAnswerQuestion {
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/question.model.ts]: all class should be in their separate files.

**Pull request merged**